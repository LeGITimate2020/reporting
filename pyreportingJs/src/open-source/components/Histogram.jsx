import React from 'react';
import PropTypes from 'prop-types';
import PlotlyComponent from './PlotlyComponentFork';

const Histogram = ({ height, values }) => {
  const layout = {
    margin: {
      t: 0,
    },
    height,
  };
  const data = [
    {
      x: values,
      type: 'histogram',
      marker: {
        color: 'rgba(200,200,250,0.7)',
      },
    },
  ];

  return <PlotlyComponent data={data} layout={layout} autoscaleVisibleYAxis={false} />;
};

Histogram.propTypes = {
  height: PropTypes.number,
  values: PropTypes.arrayOf(PropTypes.number).isRequired,
};

Histogram.defaultProps = {
  height: 200,
};

export default Histogram;
