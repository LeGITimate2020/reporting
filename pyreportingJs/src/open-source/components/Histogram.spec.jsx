import { shallow } from 'enzyme';
import React from 'react';
import Histogram from './Histogram';

describe('Histogram.jsx', () => {
  const values = [13, 23, 33];

  const props = {
    values,
  };

  it('renders without crashing', () => {
    shallow(<Histogram {...props} />);
  });

  it('Return plotly component with correct props', () => {
    const wrapper = shallow(<Histogram {...props} />);

    // Should be 1 plotly component
    expect(wrapper.find('PlotlyComponentFork').length).toBe(1);

    // props should be transformed to plotly style
    expect(wrapper.props().data[0].x).toEqual(values);
  });

  it('Matches snapshot', () => {
    const wrapper = shallow(<Histogram {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
