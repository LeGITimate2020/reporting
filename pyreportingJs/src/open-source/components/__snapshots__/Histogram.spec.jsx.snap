// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Histogram.jsx Matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Histogram
    height={200}
    values={
      Array [
        13,
        23,
        33,
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "autoscaleVisibleYAxis": false,
      "data": Array [
        Object {
          "marker": Object {
            "color": "rgba(200,200,250,0.7)",
          },
          "type": "histogram",
          "x": Array [
            13,
            23,
            33,
          ],
        },
      ],
      "layout": Object {
        "height": 200,
        "margin": Object {
          "t": 0,
        },
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "autoscaleVisibleYAxis": false,
        "data": Array [
          Object {
            "marker": Object {
              "color": "rgba(200,200,250,0.7)",
            },
            "type": "histogram",
            "x": Array [
              13,
              23,
              33,
            ],
          },
        ],
        "layout": Object {
          "height": 200,
          "margin": Object {
            "t": 0,
          },
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
